File: qc-application/src/main/java/com/eager/questioncloud/application/api/user/profile/controller/UserProfileController.java
package com.eager.questioncloud.application.api.user.profile.controller;

import com.eager.questioncloud.application.api.common.DefaultResponse;
import com.eager.questioncloud.application.api.user.profile.dto.UserPointControllerRequest.UpdateMyInformationRequest;
import com.eager.questioncloud.application.api.user.profile.dto.UserPointControllerResponse.MyInformationResponse;
import com.eager.questioncloud.application.api.user.profile.service.UserProfileService;
import com.eager.questioncloud.application.security.UserPrincipal;
import com.eager.questioncloud.core.domain.user.dto.MyInformation;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PatchMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/user/profile")
@RequiredArgsConstructor
public class UserProfileController {
    private final UserProfileService userService;

    @GetMapping("/me")
    public MyInformationResponse getMyInformation(@AuthenticationPrincipal UserPrincipal userPrincipal) {
        return new MyInformationResponse(MyInformation.from(userPrincipal.getUser()));
    }

    @PatchMapping("/me")
    public DefaultResponse updateMyInformation(
        @AuthenticationPrincipal UserPrincipal userPrincipal, @RequestBody @Valid UpdateMyInformationRequest request) {
        userService.updateUserInformation(userPrincipal.getUser(), request.getName(), request.getProfileImage());
        return DefaultResponse.success();
    }
}

File: qc-application/src/main/java/com/eager/questioncloud/application/api/user/profile/controller/UserProfileController.kt
package com.eager.questioncloud.application.api.user.profile.controller

import com.eager.questioncloud.application.api.common.DefaultResponse
import com.eager.questioncloud.application.api.common.DefaultResponse.Companion.success
import com.eager.questioncloud.application.api.user.profile.dto.MyInformationResponse
import com.eager.questioncloud.application.api.user.profile.dto.UpdateMyInformationRequest
import com.eager.questioncloud.application.api.user.profile.service.UserProfileService
import com.eager.questioncloud.application.security.UserPrincipal
import com.eager.questioncloud.core.domain.user.dto.MyInformation.Companion.from
import jakarta.validation.Valid
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/user/profile")
class UserProfileController(
    private val userService: UserProfileService
) {
    @GetMapping("/me")
    fun getMyInformation(@AuthenticationPrincipal userPrincipal: UserPrincipal): MyInformationResponse {
        return MyInformationResponse(from(userPrincipal.user))
    }

    @PatchMapping("/me")
    fun updateMyInformation(
        @AuthenticationPrincipal userPrincipal: UserPrincipal, @RequestBody request: @Valid UpdateMyInformationRequest
    ): DefaultResponse {
        userService.updateUserInformation(userPrincipal.user, request.name, request.profileImage)
        return success()
    }
}
